src/chapters/enums/intro.rst

```diff
@@ -0,0 +1,346 @@
+
+.. sourcecode:: java
+   :linenos:
+
+   import java.util.Scanner;
+
+   public class DayPrinter {
+      public static void main(String[] args) {
+         Scanner in = new Scanner(System.in);
+         System.out.println("Enter an integer: ");
+         int dayNum = in.nextInt();
+
+         String day;
+         switch (dayNum) {
+            case DayStatic.SUNDAY:
+               day = "Sunday";
+               break;
+            case DayStatic.MONDAY:
+               day = "Monday";
+               break;
+            case DayStatic.TUESDAY:
+               day = "Tuesday";
+               break;
+            case DayStatic.WEDNESDAY:
+               day = "Wednesday";
+               break;
+            case DayStatic.THURSDAY:
+               day = "Thursday";
+               break;
+            case DayStatic.FRIDAY:
+               day = "Friday";
+               break;
+            case DayStatic.SATURDAY:
+               day = "Saturday";
+               break;
+            default:
+               // in this example, this block runs if none of the above blocks match
+               day = "Int does not correspond to a day of the week";
+         }
+         System.out.println(day);
+      }
+   }
+
+In essence, this code represents days of the week as fixed integer values, one for each day. Enum types are essentially a more robust version of this approach. 
+
+Let's revisit our ``Day`` enum type:
+
+.. sourcecode:: java
+   :linenos:
+
+   public enum Day {
+   
+      SUNDAY,
+      MONDAY,
+      TUESDAY,
+      WEDNESDAY,
+      THURSDAY,
+      FRIDAY,
+      SATURDAY
+
+   }
+
+We can declare a variable of type ``Day`` and it will only be allowed to take on one of the 7 defined values.
+
+.. sourcecode:: java
+   :linenos:
+
+   // This works
+   Day workWeekStart = Day.MONDAY;
+
+   // This does not, throwing a compiler error
+   Day workWeekEnd = "Friday";
+
+Enums are important because they provide *type safety* in situations where we want to restrict possible values. In other words, they eliminate the possibility of bad, or dirty, values.
+
+Enum Examples
+-------------
+
+The world is filled with examples ripe for representation by enums. Here are a few from both the real world and the world of programming.
+
+.. admonition:: Example
+
+   Months of the year.
+
+   .. sourcecode:: java
+      :linenos:
+
+      public enum Month {
+      
+         JANUARY,
+         FEBRUARY,
+         MARCH,
+         APRIL,
+         MAY,
+         JUNE,
+         JULY,
+         AUGUST,
+         SEPTEMBER,
+         OCTOBER,
+         NOVEMBER,
+         DECEMBER
+      
+      }
+
+
+.. admonition:: Example
+
+   Given a model type like our ``Event`` class, enums can represent categories that model objects can fall into.
+
+   .. sourcecode:: java
+      :linenos:
+
+      public enum EventCategory {
+
+         CONFERENCE,
+         MEETUP,
+         WORKSHOP,
+         SOCIAL
+
+      }
+
+.. index:: log level
```

Comment:
as far as I know, this si the first time we've mentioned this term so i htink it also warrants a "!"